python/ray/tests/accelerators/test_nvidia_gpu.py

```diff
@@ -1,43 +1,209 @@
-Bus Location:    0000:00:1e.0
-Device Minor:    0
-Blacklisted:     No
-    """
-    with patch("builtins.open", mock_open(read_data=info_string)):
-        assert NvidiaGPUAcceleratorManager.get_current_node_accelerator_type() == "V100"
-
-    info_string = """Model:           Tesla T4
-IRQ:             10
-GPU UUID:        GPU-415fe7a8-f784-6e3d-a958-92ecffacafe2
-Video BIOS:      90.04.84.00.06
-Bus Type:        PCIe
-DMA Size:        47 bits
-DMA Mask:        0x7fffffffffff
-Bus Location:    0000:00:1b.0
-Device Minor:    0
-Blacklisted:     No
-    """
-    with patch("builtins.open", mock_open(read_data=info_string)):
-        assert NvidiaGPUAcceleratorManager.get_current_node_accelerator_type() == "T4"
+class DeviceHandleMock(dict):
+    def __init__(
+        self,
+        name: str,
+        uuid: str,
+        mig_devices: List["DeviceHandleMock"] = None,
+        **kwargs
+    ):
+        super().__init__()
+        self["name"] = name
+        self["uuid"] = uuid
+        if mig_devices is not None:
+            self["mig_devices"] = mig_devices
+        self.update(kwargs)
+
+
+GPU_MOCK_DATA = [
+    DeviceHandleMock(
+        "Tesla V100-SXM2-16GB", "GPU-8eaaebb8-bb64-8489-fda2-62256e821983"
+    ),
+    DeviceHandleMock(
+        "Ampere A100-SXM4-40GB",
+        "GPU-8eaaebb8-bb64-8489-fda2-62256e821983",
+        mig_devices=[
+            DeviceHandleMock(
+                "Ampere A100-SXM4-40GB MIG 1g.5gb",
+                "MIG-c6d4f1ef-42e4-5de3-91c7-45d71c87eb3f",
+                gi_id=0,
+                ci_instance=0,
+            ),
+            DeviceHandleMock(
+                "Ampere A100-SXM4-40GB MIG 1g.5gb",
+                "MIG-0c757cd7-e942-5726-a0b8-0e8fb7067135",
+                gi_id=1,
+                ci_instance=0,
+            ),
+        ],
+    ),
+    DeviceHandleMock(
+        "Ampere A100-SXM4-40GB",
+        "GPU-8eaaebb8-bb64-8489-fda2-62256e821983",
+        mig_devices=[
+            DeviceHandleMock(
+                "Ampere A100-SXM4-40GB MIG 1g.5gb",
+                "MIG-a28ad590-3fda-56dd-84fc-0a0b96edc58d",
+                gi_id=0,
+                ci_instance=0,
+            )
+        ],
+    ),
+]
+
+
+# pnvml mock for gpu resources
+class PyNVMLMock:
+    def __init__(self, mock_data=GPU_MOCK_DATA, driver_version="535.104.12"):
+        self._mock_data = mock_data
+        self.driver_version = driver_version
+
+    def nvmlInit(self):
+        return
+
+    def nvmlShutdown(self):
+        return
+
+    def nvmlSystemGetDriverVersion(self):
+        return self.driver_version
+
+    def nvmlDeviceGetCount(self):
+        return len(self._mock_data)
+
+    def nvmlDeviceGetHandleByIndex(self, index):
+        return self._mock_data[index]
+
+    def nvmlDeviceGetName(self, handle):
+        return handle.get("name", "")
+
+    def nvmlDeviceGetMaxMigDeviceCount(self, handle):
+        if "mig_devices" in handle:
+            return len(handle["mig_devices"])
```

Comment:
`nvmlDeviceGetMaxMigDeviceCount()` will return either 7 or 0 on MIG-supported devices, or raise `NVMLError_NotSupported` error for unsupported devices.